src/components/views/calendar/Calendar.tsx

```diff
@@ -1,145 +1,281 @@
+    const [day, month, year] = dateString.split("/").map(Number); // Split the date string into parts and convert to numbers
+    const timeMatch = timeString.match(/\d+/g); // Match hours and minutes from the time string
+
+    if (!timeMatch) {
+      throw new Error("Invalid time format");
+    }
+
+    const [hours, minutes] = timeMatch.map(Number); // Extract hours and minutes from the time match
+
+    const isPM = timeString.includes("PM"); // Check if the time is PM
+
+    // Adjust hours if PM (assuming 12-hour format)
+    let adjustedHours = hours;
+    if (isPM && adjustedHours !== 12) {
+      adjustedHours += 12;
+    }
+
+    // Create a new Date object with the parsed components
+    return new Date(year, month - 1, day, adjustedHours, minutes);
+  }
+
+  const startDate = parseDateStringWithTime(
+    apiEvent.attributes.start_date,
+    apiEvent.attributes.start_time
+  );
+  const endDate = parseDateStringWithTime(
+    apiEvent.attributes.end_date,
+    apiEvent.attributes.end_time
+  );
+
+  // Create and return a ScheduleEvent object
+  return {
+    Id: parseInt(apiEvent.id), // Parse the string id to number
+    Subject: apiEvent.attributes.title,
+    Description: apiEvent.attributes.description,
+    StartTime: new Date(startDate),
+    EndTime: new Date(endDate),
+  };
+};
 
 function Calendar({ user }: Props) {
   const navigate = useNavigate();
+  const [scheduleData, setScheduleData] = useState<ScheduleEvent[]>([]);
   const scheduleObj = useRef<ScheduleComponent>(null);
-//   // const storedEvents: any[] = JSON.parse(localStorage.getItem('events') || '[]');
-//   const [events, setEvents] = useState<any[]>([]);
-//   // console.log(scheduleObj)
-// //undefined
-//   // console.log(scheduleObj.current?.eventsData) //undefined
-//   // console.log(scheduleObj.current?.activeCellsData)
-//   // console.log(scheduleObj.current?.eventsProcessed)
-
-//   useEffect(() => {
-  //     setEvents([...events, newNewEvent])
-  //     }, [])
-  
-      const newNewEvent: any = {
-        dataSource: [{
-          Subject: 'Testing',
-          EndTime: new Date("2024-05-17T18:00:00.000Z"),
-          StartTime: new Date("2024-05-17T14:00:00.000Z"),
-          IsAllDay: false,
-          Description: 'Testing'
-        }]
+  const currentToken = sessionStorage.getItem("token");
+  const windowLocation = window.location.pathname;
+  console.log(windowLocation);
+
+  if (!currentToken) {
+    throw new Error("Token is null");
+  }
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await fetchCalendarEvents(user.data.id, currentToken);
+        // Transform API response to ScheduleEvent
+        const newEvents = response.data.map((event: ApiEvent) => {
+          return transformToScheduleEvent(event);
+        });
+        // Every time code is edited, this is run again duplicating on the page, maybe there is a method to catch this error.
```

Comment:
Fixed this duplication -> you were using the spread operator which in short takes the fetched data and adds to it. Instead we replace it since we are already fetching that data on the page load